covasim/sim.py

```diff
@@ -91,6 +92,15 @@ def __init__(self, pars=None, datafile=None, datacols=None, popfile=None, filena
         self.load_data(datafile, datacols) # Load the data, if provided
         self.load_population(popfile)      # Load the population, if provided
         self.update_pars(pars)             # Update the parameters, if provided
+
+        # if resampling is enabled, make 100k(default) pop size
+        if self["resample"] and self["n"] > self["resample_max_pop"]:
+            self.max_n = self["n"]
+            self["n"] = self["resample_max_pop"]
+
+        # dynamic resampling
+        self.resampling_threshold = 0.5  # Fraction of total pop that isn't susceptible when we trigger scaling
+        self.resampling_factor = 2  # Factor by which we will rescale population
```

Comment:
i think we'd want the threshold much lower, e.g. <0.1. prevalence of 50% is very different than prevalence of 0.05%. but i'm not sure this is the right denominator -- maybe we want to have a ceiling on the number of active infections instead? so if that exceeds say 5000, then we scale by `self.resampling_factor`